src/main/java/fi/aalto/cs/apluscourses/presentation/exercise/ExerciseGroupViewModel.java

```diff
@@ -20,69 +18,44 @@ public ExerciseGroupViewModel(@NotNull ExerciseGroup exerciseGroup) {
         .values()
         .stream()
         .map(ExerciseViewModel::new)
-        .sorted(EXERCISE_COMPARATOR) // O1_SPECIFIC
+        .sorted(EXERCISE_COMPARATOR)
         .collect(Collectors.toList()));
   }
 
   public String getPresentableName() {
     return APlusLocalizationUtil.getEnglishName(getModel().getName());
   }
 
-  // O1_SPECIFIC
   private static final Comparator<ExerciseViewModel> EXERCISE_COMPARATOR = (exercise1, exercise2)
       -> {
-    int chapter1 = getExerciseChapter(exercise1);
-    int chapter2 = getExerciseChapter(exercise2);
-    if (chapter1 > chapter2) {
-      return 1;
-    }
-    if (chapter2 > chapter1) {
-      return -1;
+    /*
+     * Sorting lexicographically by the HTML URL almost works, except for the assignment numbers,
+     * since "...w01_ch04_9/" > "...w01_ch04_10/". So we compare the HTML URLs without the
+     * assignment numbers, and if they aren't equal, then there is a clear "winner". If they are
+     * equal, we compare the HTML URLs by length first, so that "...10/" becomes greater than
+     * "...9/". This conveniently also makes feedback exercises the last exercises in their
+     * chapters.
+     */
```

Comment:
Are those codes like `w01_ch04_9` generated by A+ (and kinda part of the "protocol") or are they chosen by teachers of the courses? If the latter, this is still a solution specific to certain courses only, right?

I was wondering if something like [natural sort](https://blog.codinghorror.com/sorting-for-humans-natural-sort-order/) would give a generic solution to the problem. Probably an overkill, though.

Reply:
Yeah that is a good question, which I've been thinking about as well :smile: I'll try to dig deeper.

Natural sort is cool, although it would still stumble on feedback assignments I think :thinking: